src/main/java/edu/byu/cs/autograder/test/UnitTestGrader.java

```diff
@@ -0,0 +1,101 @@
+package edu.byu.cs.autograder.test;
+
+import edu.byu.cs.autograder.GradingContext;
+import edu.byu.cs.autograder.GradingException;
+import edu.byu.cs.model.Phase;
+import edu.byu.cs.model.RubricConfig;
+
+import java.io.File;
+import java.util.HashSet;
+import java.util.Set;
+
+public class UnitTestGrader extends TestGrader {
+    public UnitTestGrader(GradingContext gradingContext) {
+        super(gradingContext);
+    }
+
+    @Override
+    protected String name() {
+        return "unit";
+    }
+
+    @Override
+    protected Set<String> excludedTests() throws GradingException {
+        return testHelper.getTestFileNames(phaseTests);
+    }
+
+    @Override
+    protected File testsToCompile() {
+        return new File(gradingContext.stageRepo(), module + "/src/test/java/");
+    }
+
+    @Override
+    protected Set<String> packagesToTest() throws GradingException {
+        return switch (gradingContext.phase()) {
+            case Phase0, Phase1, Phase6 -> throw new GradingException("No unit tests for this phase");
+            case Phase3 -> Set.of("serviceTests");
+            case Phase4 -> Set.of("dataAccessTests");
+            case Phase5 -> Set.of("clientTests");
```

Comment:
That sounds reasonable. I thought most of the hardcoding was temporary, but this part may not be. Do you think it would go OK inside of PhaseUtils?

Reply:
I'm thinking it's okay if this is hardcoded, because it is very unlikely to change so long as Chess is the project, and at that point... 🤷🏻‍♂️ 

Yeah, `PhaseUtils` would probably be fine

Reply:
How's that?

Reply:
Yes, I like that a ton better! Looks great!